bugwarrior/services/github.py

```diff
@@ -278,29 +293,34 @@ def get_involved_issues(self, user):
         """ Grab all 'interesting' issues """
         issues = {}
         for issue in self.client.get_involved_issues(user):
-            url = issue['html_url']
-            tag = re.match('.*github\\.com/(.*)/(issues|pull)/[^/]*$', url)
-            if tag is None:
-                log.critical(" Unrecognized issue URL: %s.", url)
-                continue
-            issues[url] = (tag.group(1), issue)
+            try:
+                url = issue['html_url']
+                repo = self.get_repository_from_issue(issue)
+                issues[url] = (repo, issue)
+            except ValueError as e:
+                log.critical(e)
         return issues
 
     def get_directly_assigned_issues(self):
-        project_matcher = re.compile(
-            r'.*/repos/(?P<owner>[^/]+)/(?P<project>[^/]+)/.*'
-        )
         issues = {}
         for issue in self.client.get_directly_assigned_issues():
-            match_dict = project_matcher.match(issue['url']).groupdict()
-            issues[issue['url']] = (
-                '{owner}/{project}'.format(
-                    **match_dict
-                ),
-                issue
-            )
+            repos = self.get_repository_from_issue(issue)
+            issues[issue['url']] = (repos, issue)
         return issues
 
+    @staticmethod
```

Comment:
Can!  But the behavior of what happens when you call other methods in your class is different between the two.  From within a `@classmethod`, you can call other `@classmethods` by using `cls.whatever_method`, and your subclasses can override `whatever_method`.  When using `@staticmethod`, you can only call those methods by using the class name directly -- `MyClass.whatever_method`, and that prevents your subclasses from being able to alter that behavior.